d2networking/d2netpacket/packet_server_full.go

```diff
@@ -0,0 +1,44 @@
+package d2netpacket
+
+import (
+	"encoding/json"
+	"log"
+	"time"
+
+	"github.com/OpenDiablo2/OpenDiablo2/d2networking/d2netpacket/d2netpackettype"
+)
+
+// ServerFullPacket contains the current time. It is sent by the server
+// to inform a client that the server has reached the max number of allowed connections.
+type ServerFullPacket struct {
+	TS time.Time `json:"ts"`
+}
+
+// CreateServerFullPacket returns a NetPacket which declares a
+// ServerFullPacket with the current time.
+func CreateServerFullPacket() NetPacket {
+	serverClosed := ServerFullPacket{
+		TS: time.Now(),
+	}
+
+	b, err := json.Marshal(serverClosed)
+	if err != nil {
+		log.Print(err)
+	}
+
+	return NetPacket{
+		PacketType: d2netpackettype.ServerFull,
+		PacketData: b,
+	}
+}
+
+// UnmarshalServerFull unmarshalls the given data to a ServerFullPacket struct
+func UnmarshalServerFull(packet []byte) (ServerFullPacket, error) {
+	var resp ServerFullPacket
+
+	if err := json.Unmarshal(packet, &resp); err != nil {
+		return resp, err
+	}
+
+	return resp, nil
+}
```

Comment:
Isn't it possible that the packet can be empty? I'm pretty sure the client does not need the timestamp from the server. Yes, it's nitpicking, but it's about 8 Bytes saved on a packet where it really doesn't matter... Up to you...

Reply:
Fair point. Switched this to an empty packet for now. We can add metadata at a later date if needed.

Reply:
Now the comment is wrong "// ServerFullPacket contains the current time.".
Also sorry for my merge conflicts!